plugins/rest/aws.go

```diff
@@ -403,6 +417,9 @@ func signV4(req *http.Request, credService awsCredentialService, theTime time.Ti
 		if err != nil {
 			return errors.New("error getting request body: " + err.Error())
 		}
+		// Since ReadAll consumed the body ReadCloser, we must create a new ReadCloser for the request so that the
+		// subsequent read starts from the beginning
+		req.Body = ioutil.NopCloser(bytes.NewReader(body))
```

Comment:
🔍 Where's that _subsequent read_? I guess I just don't see it...

Ah, found the test. This is more of a bug fix then, I suppose, since the same issue had existed with the existing code? 

It looks like we don't actually `req.Body.Close()` anywhere. Should we?

💡 Turns out you had stated all this in the PR preamble already. Thank you!

Reply:
Yeah I think the subsequent read is likely somewhere within the http library! :)